### DISCUSSION TITLE
Tests and next/router

### DISCUSSION THREAD

**Discussion by UniversalTraveller** [neuefische/questions#142] [Web]
When I test my components that use routing, I get this error message:

`TypeError: Cannot read properties of null (reading 'query')`

In this thread I found a possible solution:
https://github.com/orgs/neuefische/discussions/110

I create a mock in my .spec.js with this code:

```
jest.mock('next/router', () => ({
	useRouter() {
		return {
			path: '/podcast/[podcast]',
			asPath: '/podcast/podcast',
			query: {
				id: '1',
			},
		};
	},
}));
```

A similar approach works just fine for Storybook. To display the same component in storybook I use this:
```

 parameters: {
		nextRouter: {
			path: '/podcast/[podcast]',
			asPath: '/podcast/podcast',
			query: {
				id: '1',
			},
		},
	},
```

For testing it doesn't really work. I just get a different error message:  `TestingLibraryElementError: Unable to find an element with the text: CSS Proposals @when, CSS Masonry, Carets. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.`

**Comment by alexclaes** [comment]
You are using the right approach!

I found one tiny, but very important but in your code, that might help you with some errors

Your mock for useRouter defines the query ID as a string:
```
query: {
    id: '1',
},
```

Your mock for the podcasts  defines the `key` as a number:
```
initialState.podcasts = [
    {
        title: 'Title',
	author: 'Author',
	key: 1, // <-------------------
	tags: ['web development', 'news', 'tech'],
    },
];
```

Therefore the code selecting an individual podcast will not work:

```js
const selectedPodcast = podcasts.find(podcast => podcast.key === id);
```
